omod/src/main/java/org/openmrs/module/reportingrest/web/resource/ReportDesignResource.java

```diff
@@ -0,0 +1,125 @@
+import org.openmrs.module.reporting.report.service.ReportService;
+import org.openmrs.module.reportingrest.web.controller.ReportingRestController;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.DefaultRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.FullRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.RefRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.EmptySearchResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+import java.util.List;
+
+/**
+ * {@link Resource} for {@link ReportDesign}s, supporting standard CRUD operations
+ */
+@Resource(name = RestConstants.VERSION_1 + ReportingRestController.REPORTING_REST_NAMESPACE + "/reportDesign",
+    supportedClass = ReportDesign.class, supportedOpenmrsVersions = {"1.8.* - 9.9.*"})
+public class ReportDesignResource extends DelegatingCrudResource<ReportDesign> {
+
+  @Override
+  public ReportDesign newDelegate() {
+    return new ReportDesign();
+  }
+
+  @Override
+  public ReportDesign getByUniqueId(String uuid) {
+    return getReportService().getReportDesignByUuid(uuid);
+  }
+
+  @Override
+  protected PageableResult doSearch(RequestContext context) {
+    String reportDefinitionUuid = context.getParameter("reportDefinitionUuid");
+    if (StringUtils.isBlank(reportDefinitionUuid)) {
+      throw new IllegalArgumentException("reportDefinitionUuid parameter is required");
+    }
+
+    ReportDefinition reportDefinition = getReportDefinitionService().getDefinitionByUuid(reportDefinitionUuid);
+    if (reportDefinition == null) {
+      return new EmptySearchResult();
+    }
+
+    List<ReportDesign> reportDesigns = getReportService().getReportDesigns(reportDefinition, null, false);
+    return new NeedsPaging<ReportDesign>(reportDesigns, context);
+  }
+
+  @Override
+  protected void delete(ReportDesign reportDesign, String reason, RequestContext requestContext) throws ResponseException {
+    purge(reportDesign, requestContext);
+  }
+
+  @Override
+  public ReportDesign save(ReportDesign reportDesign) {
+    return getReportService().saveReportDesign(reportDesign);
+  }
+
+  @Override
+  public void purge(ReportDesign reportDesign, RequestContext requestContext) throws ResponseException {
+    if (reportDesign != null) {
```

Comment:
Is there a use case where reportDesign would be null here?

Reply:
probably not, corrected